server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBPLModifyHelper.java

```diff
@@ -0,0 +1,194 @@
+import org.springframework.context.annotation.Lazy;
+import org.springframework.integration.jdbc.lock.JdbcLockRegistry;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
+
+import com.oceanbase.odc.common.util.StringUtils;
+import com.oceanbase.odc.common.util.VersionUtils;
+import com.oceanbase.odc.core.authority.util.SkipAuthorize;
+import com.oceanbase.odc.core.session.ConnectionSession;
+import com.oceanbase.odc.core.session.ConnectionSessionConstants;
+import com.oceanbase.odc.core.session.ConnectionSessionUtil;
+import com.oceanbase.odc.core.shared.PreConditions;
+import com.oceanbase.odc.core.shared.constant.ErrorCodes;
+import com.oceanbase.odc.core.shared.constant.LimitMetric;
+import com.oceanbase.odc.core.shared.exception.BadRequestException;
+import com.oceanbase.odc.core.shared.exception.ConflictException;
+import com.oceanbase.odc.core.sql.execute.SyncJdbcExecutor;
+import com.oceanbase.odc.core.sql.execute.model.SqlTuple;
+import com.oceanbase.odc.core.sql.split.OffsetString;
+import com.oceanbase.odc.service.common.util.SqlUtils;
+import com.oceanbase.odc.service.connection.model.ConnectionConfig;
+import com.oceanbase.odc.service.db.model.EditPLReq;
+import com.oceanbase.odc.service.db.model.EditPLResp;
+import com.oceanbase.odc.service.session.ConnectConsoleService;
+import com.oceanbase.odc.service.session.SessionProperties;
+import com.oceanbase.odc.service.session.interceptor.SqlCheckInterceptor;
+import com.oceanbase.odc.service.session.interceptor.SqlConsoleInterceptor;
+import com.oceanbase.odc.service.session.interceptor.SqlExecuteInterceptorService;
+import com.oceanbase.odc.service.session.model.AsyncExecuteContext;
+import com.oceanbase.odc.service.session.model.SqlAsyncExecuteReq;
+import com.oceanbase.odc.service.session.model.SqlAsyncExecuteResp;
+import com.oceanbase.tools.dbbrowser.model.DBObjectType;
+
+import lombok.NonNull;
+
+/**
+ * @description:
+ * @author: zijia.cj
+ * @date: 2024/10/17 18:59
+ * @since: 4.3.3
+ */
+@Validated
+@Component
+@SkipAuthorize("permission check inside")
+public class DBPLModifyHelper {
+    public static final String ODC_TEMPORARY_PROCEDURE = "_ODC_TEMPORARY_PROCEDURE";
+    public static final String ODC_TEMPORARY_TRIGGER = "_ODC_TEMPORARY_TRIGGER";
+    public static final String ODC_TEMPORARY_FUNCTION = "_ODC_TEMPORARY_FUNCTION";
+    public static final String OB_VERSION_SUPPORT_MULTIPLE_SAME_TRIGGERS = "4.2";
+    public static final int OBTAIN_LOCK_TIME = 3;
+    @Autowired
+    @Lazy
+    private ConnectConsoleService connectConsoleService;
+    @Autowired
+    private JdbcLockRegistry jdbcLockRegistry;
+    @Autowired
+    private SessionProperties sessionProperties;
+    @Autowired
+    private SqlExecuteInterceptorService sqlInterceptService;
+
+    public EditPLResp editPL(@NotNull ConnectionSession connectionSession, @NotNull @Valid EditPLReq editPLReq,
+            boolean needSqlRuleCheck)
```

Comment:
Why does it need the `boolean needSqlRuleCheck` parameter?

Reply:
why？（only test）